src/test/java/seedu/address/logic/commands/DeleteAssignmentCommandTest.java

```diff
@@ -0,0 +1,213 @@
+
+        assertTrue(model.getFilteredPersonList().isEmpty());
+    }private Model model = new ModelManager(getTypicalAddressBook(), new UserPrefs());
+
+    @Test
+    public void execute_validIndexUnfilteredList_success() {
+        Person personToDelete = model.getFilteredPersonList().get(INDEX_FIRST_PERSON.getZeroBased());
+        DeletePersonCommand deletePersonCommand = new DeletePersonCommand(INDEX_FIRST_PERSON);
+
+        String expectedMessage = String.format(DeletePersonCommand.MESSAGE_DELETE_PERSON_SUCCESS, personToDelete);
+
+        ModelManager expectedModel = new ModelManager(model.getAddressBook(), new UserPrefs());
+        expectedModel.deletePerson(personToDelete);
+
+        assertCommandSuccess(deletePersonCommand, model, expectedMessage, expectedModel);
+    }
+
+    @Test
+    public void execute_invalidIndexUnfilteredList_throwsCommandException() {
+        Index outOfBoundIndex = Index.fromOneBased(model.getFilteredAssignmentList().size() + 1);
+        DeleteAssignmentCommand deleteAssignmentCommand =
+                new DeleteAssignmentCommand(new PersonBuilder().build().getName(), outOfBoundIndex);
+
+        assertCommandFailure(deleteAssignmentCommand, model, Messages.MESSAGE_INVALID_ASSIGNMENT_DISPLAYED_INDEX);
+    }
+
+    @Test
+    public void execute_validIndexFilteredList_success() {
+        showAssignmentAtIndex(model, INDEX_FIRST_ASSIGNMENT);
+
+        Assignment assignmentToDelete = model.getFilteredAssignmentList().get(INDEX_FIRST_ASSIGNMENT.getZeroBased());
+        DeleteAssignmentCommand deleteAssignmentCommand =
+                new DeleteAssignmentCommand(new PersonBuilder().build().getName(), INDEX_FIRST_ASSIGNMENT);
+
+        String expectedMessage = String.format(DeleteAssignmentCommand.MESSAGE_DELETE_ASSIGNMENT_SUCCESS,
+         assignmentToDelete);
+
+        Model expectedModel = new ModelManager(model.getAddressBook(), new UserPrefs());
+        expectedModel.deleteAssignment(new PersonBuilder().build(), assignmentToDelete);
+        showNoAssignment(expectedModel);
+
+        assertCommandSuccess(deleteAssignmentCommand, model, expectedMessage, expectedModel);
+    }
+    */
+
+
+    /* @Test
+    public void execute_invalidIndexFilteredList_throwsCommandException() {
+        showAssignmentAtIndex(model,
+                model.getFilteredPersonList().get(INDEX_FIRST_PERSON.getZeroBased()), INDEX_FIRST_ASSIGNMENT);
+
+        Index outOfBoundIndex = INDEX_SECOND_ASSIGNMENT;
+        // ensures that outOfBoundIndex is still in bounds of address book list
+        assertTrue(outOfBoundIndex.getZeroBased() < model.getAddressBook().getAssignmentsList().size());
+
+        DeleteAssignmentCommand deleteAssignmentCommand =
+                new DeleteAssignmentCommand(new PersonBuilder().build().getName(), outOfBoundIndex);
+
+        assertCommandFailure(deleteAssignmentCommand, model, Messages.MESSAGE_INVALID_ASSIGNMENT_DISPLAYED_INDEX);
+    }
+
+    */
```

Comment:
Same as first comment

Reply:
for further modification, should be included

Reply:
Alright noted!